### PR TITLE
Update bucket attributes in order to disable soft delete

### PR DESCRIPTION
**Pull request by ycohen278**
update bucket attributes to disable soft delete. it does not work when setting it when creating the bucket.

### CODE REVIEW

File: cli_tools/common/utils/storage/scratch_bucket_creator.go [hunk 1]
```diff
@@ -134,16 +132,30 @@ func (c *ScratchBucketCreator) createBucketIfNotExisting(project string,
...
+			return "", err
+		}
 	}
-	log.Printf("Creating scratch bucket `%v` in %v region", bucketAttrs.Name, bucketAttrs.Location)
-	if err := c.StorageClient.CreateBucket(bucketAttrs.Name, project, bucketAttrs); err != nil {
+	log.Printf("Updating soft delete property of scratch bucket `%v` in %v region", bucketAttrs.Name, bucketAttrs.Location)
+	if err = c.removeSoftDeleteFromBucket(bucketAttrs.Name); err != nil {
 		return "", err
 	}
+	if foundBucketAttrs != nil {
```
Comment (advazair) [inline cli_tools/common/utils/storage/scratch_bucket_creator.go:146]: why is it needed?

File: cli_tools/common/utils/storage/scratch_bucket_creator.go [hunk 2]
```diff
@@ -134,16 +132,30 @@ func (c *ScratchBucketCreator) createBucketIfNotExisting(project string,
 	if err != nil {
```
Comment (advazair) [inline cli_tools/common/utils/storage/scratch_bucket_creator.go:132]: you can move to line 80, then you can omit the "if err == nil" statement.

File: cli_tools/common/utils/storage/scratch_bucket_creator.go [hunk 3]
```diff
@@ -81,9 +81,7 @@ func (c *ScratchBucketCreator) getBucketAttrs(fileGcsPath string, project string
 	if err == nil {
```
Comment (advazair) [inline cli_tools/common/utils/storage/scratch_bucket_creator.go:81]: Change to "if err != nil" then return nil, err

File: cli_tools/common/utils/storage/scratch_bucket_creator.go [hunk 4]
```diff
@@ -134,16 +132,30 @@ func (c *ScratchBucketCreator) createBucketIfNotExisting(project string,
 	if err != nil {
 		return "", err
 	}
-	if foundBucketAttrs != nil {
-		return foundBucketAttrs.Location, nil
+
+	if foundBucketAttrs == nil {
```
Comment (advazair) [inline cli_tools/common/utils/storage/scratch_bucket_creator.go:136]: Why are we creating a new bucket when the bucket attributes are nil? Does it mean the bucket doesn't exist? Shouldn't the client return an error in this case?

File: cli_tools/common/utils/storage/scratch_bucket_creator.go (hunk 1 above)
Comment (ycohen278) [reply cli_tools/common/utils/storage/scratch_bucket_creator.go:147]: i think they can, there is an option for the bucket to be supplied by the user, dont want to change current behavior.

File: cli_tools/common/utils/storage/scratch_bucket_creator.go (hunk 4 above)
Comment (ycohen278) [reply cli_tools/common/utils/storage/scratch_bucket_creator.go:137]: this is what this method does, it checks if the bucket already exist, if not it creates a new one.

**Review by advazair — APPROVED** [review]

**Pull request merged**